packages/components/carousel/src/Carousel.tsx

```diff
@@ -1,89 +1,162 @@
-        {...props}
-        aria-label="Image and video carousel"
-        renderIndicator={(
-          clickHandler: MouseEventHandler<HTMLButtonElement>,
-          isSelected: boolean,
-          index: number,
-        ) => (
-          <IndicatorItem key={index}>
-            <IndicatorButton
-              type="button"
-              aria-pressed={isSelected}
-              aria-label={`Slide ${index + 1}`}
-              onClick={clickHandler}
-              isSelected={isSelected}
-            >
-              {isSelected ? '●' : '○'}
-            </IndicatorButton>
-          </IndicatorItem>
-        )}
+    <CarouselStyled
+      styled={mergedThemeValues as Required<CarouselStyledProps>}
+      {...props}
+    >
+      <DraggableArea
+        data-testid="draggable-area"
+        onMouseDown={(event: React.MouseEvent<HTMLDivElement, MouseEvent>) =>
+          utilsHandleDragStart(event, dragStartX)
+        }
+        onMouseUp={(event: React.MouseEvent<HTMLDivElement, MouseEvent>) =>
+          utilsHandleDragEnd(event, dragStartX, nextSlide, prevSlide)
+        }
+      />
+      <CarouselTrack
+        styled={mergedThemeValues as Required<CarouselStyledProps>}
+        totalSlides={infiniteLoop ? totalSlides + 2 : totalSlides}
+        currentSlide={currentSlide}
+        visibleSlides={visibleSlides}
+        onTransitionEnd={handleTransitionEnd}
+        ref={trackRef}
       >
-        {contents.map((content: Content, index: number) => (
-          <Slide key={`carousel-container-${index}`}>
-            {renderContent(content)}
-          </Slide>
+        {infiniteLoop && (
+          <CarouselSlide key={`clone-last-${slides[totalSlides - 1].children}`}>
+            {slides[totalSlides - 1].children}
+          </CarouselSlide>
+        )}
+        {slides.map((slide: CarouselSlideProps, index: number) => (
+          <CarouselSlide key={`slide-${index}`}>{slide.children}</CarouselSlide>
         ))}
-      </ResponsiveCarousel>
+        {infiniteLoop && (
+          <CarouselSlide key={`clone-first-${slides[0].children}`}>
+            {slides[0].children}
+          </CarouselSlide>
+        )}
+      </CarouselTrack>
+
+      {showArrows && (
+        <>
+          <ArrowButton
+            styled={mergedThemeValues as Required<CarouselStyledProps>}
+            direction="left"
+            role="button"
+            aria-label="previous slide"
+            onClick={prevSlide}
+          >
+            <Icon name="chevron-up" />
+          </ArrowButton>
+          <ArrowButton
+            styled={mergedThemeValues as Required<CarouselStyledProps>}
+            direction="right"
+            role="button"
+            aria-label="next slide"
+            onClick={nextSlide}
+          >
+            <Icon name="chevron-down" />
+          </ArrowButton>
+        </>
+      )}
+
+      {showIndicators && (
+        <IndicatorWrapper>
+          {Array.from({ length: totalSlides }).map(
```

Comment:
why are you doing this instead of using a for loop?

Reply:
better integration with jsx and better functional aproach since it doesn't modify the original array